transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/httpx/OpeningHandshakeIT.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2007-2015, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.transport.wsn.specification.httpx;
+
+import static org.kaazing.test.util.ITUtil.createRuleChain;
+
+import java.net.URI;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.kaazing.gateway.server.test.GatewayRule;
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+
+public class OpeningHandshakeIT {
+    private final K3poRule k3po = new K3poRule().setScriptRoot("org/kaazing/specification/httpx/opening");
+
+    private GatewayRule gateway = new GatewayRule() {
+        {
+            GatewayConfiguration configuration =
+                    new GatewayConfigurationBuilder()
+                        .service()
+                            .accept(URI.create("ws://localhost:8080/path"))
+                            .type("echo")
+                            .crossOrigin()
+                                .allowOrigin("*")
+                            .done()
+                        .done()
+                    .done();
+            init(configuration);
+        }
+    };
+
+    @Rule
+    public TestRule chain = createRuleChain(gateway, k3po);
+
+    @Test
+    @Specification({"extended.handshake.protocol.not.negotiated/request"})
```

Comment:
This test is not relevant for the server, although it happens to pass because the client script does not verify that the Sec-WebSocket-Protocol is not present ( and there is no reason why it should). I have been following the convention of making client only test methods package private without the @Test tag. That makes it easy to see them as being distinct in eclipse, but also keeps them as methods in the file so it's easy to check that all methods from the spec test have been taken into account.